README.md
# TYPO3 Extension begroups_roles

[![Latest Stable Version](https://img.shields.io/packagist/v/ichhabrecht/begroups-roles.svg)](https://packagist.org/packages/ichhabrecht/begroups-roles)
[![Build Status](https://img.shields.io/travis/IchHabRecht/begroups_roles/master.svg)](https://travis-ci.org/IchHabRecht/begroups_roles)
[![StyleCI](https://styleci.io/repos/18537810/shield?branch=master)](https://styleci.io/repos/18537810)

Use backend user groups as switchable roles

![Role switcher](Documentation/Images/role_switcher.png)

## Installation

Simply install the extension with Composer or the Extension Manager.

```
composer require ichhabrecht/begroups-roles
```

## Usage

1. Add multiple backend groups, each for one purpose
   - Tick the checkbox `Use this group as role` 
   - Limit the modules, tables and database mount to the purpose
   
2. For convenience add the created groups to a parent group 

3. Assign the created (parent) group to backend users
   - Tick the checkbox `Use groups as roles`
   - To allow only one role group simultaneously, tick the checkbox `Restrict to one group`


ChangeLog
2020-04-22         [RELEASE] Release of begroups_roles 3.1.2 (Nicole Cordes)
2020-04-22 f3c1969 [BUGFIX] Preserve usergroup ordering for group detection (Nicole Cordes)

2020-04-22         [RELEASE] Release of begroups_roles 3.1.1 (Nicole Cordes)
2020-04-22 afd4476 [TASK] Raise compatibility to TYPO3 10.4 (Nicole Cordes)

2020-04-06         [RELEASE] Release of begroups_roles 3.1.0 (Nicole Cordes)
2020-04-06 d661586 [FEATURE] Add compatibility to TYPO3 10.3 (Nicole Cordes)
2020-04-03 ae9a3e3 [FEATURE] Add compatibility to TYPO3 9.5 (Nicole Cordes)
2020-04-03 a35c178 [TASK] Update supporting files to EXT:my_skeleton 0.3.0 (Nicole Cordes)
2018-02-01 3fcfc21 [FEATURE] Add further tests for RoleSwitcher (Nicole Cordes)
2018-02-01 9223bb3 [FEATURE] SonarQube: Add PHPUnit test and coverage results import (Nicole Cordes)
2018-02-01 9ced92c [FEATURE] Run functional tests on Travis CI (Nicole Cordes)
2018-02-01 5a9f14d [FEATURE] Add basic function tests for RoleSwitcher (Nicole Cordes)
2018-02-01 3746e70 [TASK] Update composer.json license definition (Nicole Cordes)
2018-02-01 2abab70 [TASK] Exclude .Build folder in .gitignore (Nicole Cordes)
2018-01-14 ff9cf30 [BUGFIX] Remove code complexity according to SonarQube Scanner (Nicole Cordes)

2018-01-14         [RELEASE] Release of begroups_roles 3.0.0 (Nicole Cordes)
2018-01-14 0efbf5a [TASK] Add README.md (Nicole Cordes)
2018-01-14 63421fd [!!!][FEATURE] Support groups and subgroups (Nicole Cordes)
2018-01-14 3c17183 [BUGFIX] Add missing Composer configuration (Nicole Cordes)
2018-01-14 90befd4 [FEATURE] Add stages to Travis CI configuration (Nicole Cordes)
2018-01-14 9107ba0 [TASK] Add more coding style rules (Nicole Cordes)
2018-01-14 4a2a630 [!!!][TASK] Change vendor name (Nicole Cordes)
2017-11-30 6f1f4e4 [FEATURE] Add context-sensitive help labels (Nicole Cordes)
2017-10-29 fba0bba [FEATURE] Prevent "All groups" in drop down (Nicole Cordes)
2017-10-29 4b39a63 [FEATURE] Use first group if tx_begroupsroles_limit is enabled (Nicole Cordes)
2017-10-29 7eef639 [FEATURE] Add new configuration option "limit" to be_users TCA (Nicole Cordes)
2017-10-29 da2d50f [BUGFIX] Correct "default" TCA configuration key (Nicole Cordes)

2017-04-28         [RELEASE] Release of begroups_roles 2.0.3 (Nicole Cordes)
2017-04-27 7e8b75b [BUGFIX] Ensure top window location is reloaded after role switching (Nicole Cordes)
2017-04-27 312c99b [BUGFIX] Ensure correct page access for admin users (Nicole Cordes)
2017-04-27 25cb34a [TASK] Raise compatibility to TYPO3 8.7 (Nicole Cordes)
2017-04-27 8079077 [FEATURE] Add .travis.yml (Nicole Cordes)
2017-04-27 66da844 [TASK] Fix coding style according to php-cs-fixer (Nicole Cordes)
2017-04-27 f8255c1 [FEATURE] Add StyleCI Coding Style Service (Nicole Cordes)
2017-04-27 a859f32 [FEATURE] Add friendsofphp/php-cs-fixer as suggestion (Nicole Cordes)
2017-04-27 e4df57a [TASK] Exclude local .php_cs files in .gitignore (Nicole Cordes)
2017-04-27 f50d55f [BUGFIX] Correction of directory name for js-files (Christian Matthes)

2016-12-31         [RELEASE] Release of begroups_roles 2.0.2 (Nicole Cordes)
2016-12-31 c6cb867 [TASK] Raise compatibility to TYPO3 8.5 (Nicole Cordes)

2016-10-29         [RELEASE] Release of begroups_roles 2.0.1 (Nicole Cordes)
2016-10-29 270d2a2 [TASK] Raise compatibility to TYPO3 8.4.99 (Nicole Cordes)

2016-09-01         [RELEASE] Release of begroups_roles 2.0.0 (Nicole Cordes)
2016-09-01 469244c [TASK] Add composer.json (Nicole Cordes)
2016-08-31 91478bb [FEATURE] Add switch for admin users (Nicole Cordes)
2016-08-31 8050830 [TASK] Namespace AbstractUserAuthentication hook (Nicole Cordes)
2016-08-30 0c78dab [BUGFIX] Use raw user record for group fetching (Nicole Cordes)
2016-08-30 4603763 [TASK] Add Ajax route and RequireJsModule (Nicole Cordes)
2016-08-30 68c119e [TASK] Show only inactive roles (Nicole Cordes)
2016-08-30 f63f2b9 [TASK] Use new API for ToolbarItems (Nicole Cordes)
2016-08-30 6f269ad [TASK] Move TCA to Configuration folder (Nicole Cordes)
2016-08-30 fc48a45 [TASK] Raise TYPO3 version to 7.6.0 (Nicole Cordes)
2016-08-30 5101382 [TASK] Set version to 2.0.0-dev (Nicole Cordes)

2014-04-09         [RELEASE] Release of begroups_roles 1.1.0 (Nicole Cordes)
2014-04-10 ddda816 [FEATURE] Add option to choose all groups (Nicole Cordes)
2014-04-08 8ec4601 [BUGFIX] Add type casting for incoming user group (Nicole Cordes)

2014-04-08         [RELEASE] Release of begroups_roles 1.0.0 (Nicole Cordes)


ext_emconf.php
<?php

/***************************************************************
 * Extension Manager/Repository config file for ext "begroups_roles".
 *
 * Auto generated 22-04-2020 17:48
 *
 * Manual updates:
 * Only the data in the array - everything else is removed by next
 * writing. "version" and "dependencies" must not be touched!
 ***************************************************************/

$EM_CONF[$_EXTKEY] = array (
  'title' => 'Backend user roles',
  'description' => 'Use backend user groups as switchable roles',
  'category' => 'module',
  'author' => 'Nicole Cordes',
  'author_email' => 'typo3@cordes.co',
  'author_company' => 'CPS-IT GmbH | biz-design.biz',
  'state' => 'stable',
  'uploadfolder' => 0,
  'createDirs' => '',
  'clearCacheOnLoad' => 1,
  'version' => '3.1.2',
  'constraints' => 
  array (
    'depends' => 
    array (
      'typo3' => '8.7.0-10.4.99',
    ),
    'conflicts' => 
    array (
    ),
    'suggests' => 
    array (
    ),
  ),
  '_md5_values_when_last_written' => 'a:28:{s:9:"ChangeLog";s:4:"2da3";s:7:"LICENSE";s:4:"b234";s:9:"README.md";s:4:"fdb8";s:13:"composer.json";s:4:"f39f";s:13:"composer.lock";s:4:"48af";s:12:"ext_icon.gif";s:4:"c9bc";s:14:"ext_tables.php";s:4:"597c";s:14:"ext_tables.sql";s:4:"3af5";s:16:"phpunit.xml.dist";s:4:"041c";s:24:"sonar-project.properties";s:4:"bb3d";s:45:"Classes/Backend/ToolbarItems/RoleSwitcher.php";s:4:"2fc6";s:35:"Classes/Hook/SwitchUserRoleHook.php";s:4:"ec1e";s:36:"Configuration/Backend/AjaxRoutes.php";s:4:"ae2a";s:41:"Configuration/TCA/Overrides/be_groups.php";s:4:"c5ca";s:40:"Configuration/TCA/Overrides/be_users.php";s:4:"2c9d";s:38:"Documentation/Images/role_switcher.png";s:4:"b8ab";s:27:"Resources/Private/.htaccess";s:4:"5cc8";s:46:"Resources/Private/Language/de.locallang_be.xlf";s:4:"5211";s:47:"Resources/Private/Language/de.locallang_csh.xlf";s:4:"b361";s:46:"Resources/Private/Language/de.locallang_db.xlf";s:4:"6732";s:43:"Resources/Private/Language/locallang_be.xlf";s:4:"2da4";s:44:"Resources/Private/Language/locallang_csh.xlf";s:4:"d279";s:43:"Resources/Private/Language/locallang_db.xlf";s:4:"1b28";s:42:"Resources/Public/Icons/SwitchUserGroup.png";s:4:"37b0";s:51:"Resources/Public/JavaScript/Toolbar/RoleSwitcher.js";s:4:"aba9";s:58:"Tests/Functional/Backend/ToolbarItems/RoleSwitcherTest.php";s:4:"be5e";s:48:"Tests/Functional/Fixtures/Database/be_groups.xml";s:4:"34d6";s:47:"Tests/Functional/Fixtures/Database/be_users.xml";s:4:"3671";}',
);

